### PR TITLE
Improve usability of modelardb_client

### PR DESCRIPTION
**Pull request by skejserjensen**
This PR refactor `modelardb_client` to simplify the code and adds the following functionality:
- Do not ask for confirmation for printing each batch if `stdout` is piped.
- Print helpful help if \h is executed.
- Flush memory if \f is executed.
- Flush edge if \F is executed.
- Support `INSERT INTO` with the only limitation being that values for generated columns must be provided.
- Print the time required to run each command instead of adding `\timing`.

Each piece of functionality is added by a separate commit so it may be simpler to review this PR one commit at a time.

### CODE REVIEW

File: crates/modelardb_client/src/main.rs [hunk 1]
```diff
@@ -214,56 +196,69 @@ fn repl(
...
+    println!("Time: {:?}\n", start_time.elapsed());
 }
 
-/// Execute an action. Currently, only the action `CommandStatementUpdate` is supported, which
-/// executes a SQL query that does not return a result on the server. The function returns [`Error`]
-/// if the action could not be executed.
-fn execute_action(
-    runtime: &Runtime,
+/// Execute an action. Currently, the following actions are supported:
+/// * `CommandStatementUpdate` executes a SQL query that does not return a result on the server.
```
Comment (CGodiksen) [inline crates/modelardb_client/src/main.rs:230]: In most places these lists are formatted with a `:` after the list item and then the first word of the description is capitalized. Maybe also use this format here.

File: crates/modelardb_client/src/main.rs [hunk 2]
```diff
@@ -214,56 +196,69 @@ fn repl(
...
         || action_command_or_query.starts_with("EXPLAIN")
+        || action_command_or_query.starts_with("SELECT")
     {
-        execute_query_and_print_result(runtime, flight_service_client, action_command_or_query)?;
+        execute_query_and_print_result(flight_service_client, action_command_or_query).await
     } else {
-        execute_action(runtime, flight_service_client, action_command_or_query)?;
+        execute_action(
+            flight_service_client,
+            "CommandStatementUpdate",
```
Comment (CGodiksen) [inline crates/modelardb_client/src/main.rs:215]: Is it intentional that the only action that can be executed using this function is `CommandStatementUpdate`? If so, maybe specify this in the function doc comment.

File: crates/modelardb_client/src/main.rs [hunk 3]
```diff
@@ -272,38 +267,55 @@ fn execute_command(
         .next()
         .ok_or("no command was provided")?
     {
+        //Print the schema of a table on the server.
```
Comment (CGodiksen) [inline crates/modelardb_client/src/main.rs:270]: Missing space after `//`.

File: crates/modelardb_client/src/main.rs [hunk 4]
```diff
@@ -272,38 +267,55 @@ fn execute_command(
...
+                for (metadata_name, metadata_value) in field.metadata() {
+                    print!(", {} {}", metadata_name, metadata_value);
                 }
-                Ok(())
-            })
+                println!();
+            }
+            Ok(())
         }
+        //Print the name of the tables on the server.
```
Comment (CGodiksen) [inline crates/modelardb_client/src/main.rs:291]: Missing space after `//`.

File: crates/modelardb_client/src/main.rs [hunk 5]
```diff
@@ -272,38 +267,55 @@ fn execute_command(
...
                     println!("{table}");
                 }
             }
             Ok(())
         }
+        // Flushes all data the server currently has in memory to disk.
+        "\\f" => execute_action(flight_service_client, "FlushMemory", "").await,
+        // Flushes all data the server currently has in memory and disk to the object store.
+        "\\F" => execute_action(flight_service_client, "FlushEdge", "").await,
+        //Print helpful information, explanations with \\ must indented more to be aligned.
```
Comment (CGodiksen) [inline crates/modelardb_client/src/main.rs:304]: Missing space after `//`.

Comment (CGodiksen) [reply crates/modelardb_client/src/main.rs:304]: Maybe also `must be indented more` instead.

**Review by CGodiksen — CHANGES_REQUESTED** [review]

File: docs/user/README.md [hunk 6]
```diff
@@ -119,7 +119,8 @@ modelardb
 
 Be aware that the REPL currently does not support splitting SQL statements over multiple lines and that the SQL
 statements do not need to end with a `;`. In addition to SQL statements, the REPL also supports listing all tables
-using `\dt` and printing the schema of a table using `\d table_name`:
+using `\dt`, printing the schema of a table using `\d table_name`, flushing data in memory to disk using `\f`,flushing
```
[the remark this answers is not in the document]
Comment (skejserjensen) [reply docs/user/README.md:122]: Fixed.

File: crates/modelardb_server/src/query/storage_engine_data_sink.rs [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    /// Return a snapshot of the set of metrics being collected by the [`DataSink`].
+    fn metrics(&self) -> Option<MetricsSet> {
+        None
+    }
+
+    async fn write_all(
```
[the remark this answers is not in the document]
Comment (skejserjensen) [reply crates/modelardb_server/src/query/storage_engine_data_sink.rs:67]: Fixed.

File: crates/modelardb_server/src/query/storage_engine_data_sink.rs [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+        Ok(data_points_inserted)
+    }
+}
+
+impl Debug for StorageEngineDataSink {
+    /// Write a string-based representation of the [`DataSink`] to `f`. Returns
+    /// `Err` if `std::write` cannot format the string and write it to `f`.
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "StorageEngineDataSink")
```
[the remark this answers is not in the document]
Comment (skejserjensen) [reply crates/modelardb_server/src/query/storage_engine_data_sink.rs:97]: Fixed.

File: crates/modelardb_server/src/query/storage_engine_data_sink.rs [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+    }
+}
+
+impl DisplayAs for StorageEngineDataSink {
+    /// Write a string-based representation of the [`DataSink`] to `f`. Returns
+    /// `Err` if `std::write` cannot format the string and write it to `f`.
+    fn fmt_as(&self, _t: DisplayFormatType, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "StorageEngineDataSink")
```
[the remark this answers is not in the document]
Comment (skejserjensen) [reply crates/modelardb_server/src/query/storage_engine_data_sink.rs:105]: Fixed.

File: crates/modelardb_server/src/query/storage_engine_data_sink.rs [hunk 10]
```diff
@@ -0,0 +1,107 @@
+/* Copyright 2023 The ModelarDB Contributors
```
[the remark this answers is not in the document]
Comment (skejserjensen) [reply crates/modelardb_server/src/query/storage_engine_data_sink.rs:1]: Fixed.

File: crates/modelardb_client/src/main.rs [hunk 11]
```diff
@@ -131,18 +128,13 @@ fn parse_command_line_arguments(mut args: Args) -> (Option<String>, Option<u16>,
...
+async fn connect(host: &str, port: u16) -> Result<FlightServiceClient<Channel>, Box<dyn Error>> {
     let address = format!("grpc://{host}:{port}");
-    runtime.block_on(async { Ok(FlightServiceClient::connect(address).await?) })
+    Ok(FlightServiceClient::connect(address).await?)
 }
 
 /// Execute the actions, commands, and queries in the file at `query_file_path`.
-fn file(
-    runtime: Runtime,
+async fn file(
```
[the remark this answers is not in the document]
Comment (skejserjensen) [reply crates/modelardb_client/src/main.rs:137]: Fixed.

File: crates/modelardb_client/src/main.rs [hunk 12]
```diff
@@ -161,29 +153,20 @@ fn file(
...
         }
     }
 
     Ok(())
 }
 
 /// Execute actions, commands and queries in a read-eval-print loop.
-fn repl(
-    runtime: Runtime,
+async fn repl(
```
[the remark this answers is not in the document]
Comment (skejserjensen) [reply crates/modelardb_client/src/main.rs:164]: Fixed.

File: crates/modelardb_client/src/main.rs [hunk 13]
```diff
@@ -214,56 +196,69 @@ fn repl(
...
+            action_command_or_query,
+        )
+        .await
+    };
+
+    if let Err(message) = result {
+        eprintln!("{message}");
     }
-    Ok(())
+    println!();
```
[the remark this answers is not in the document]
Comment (skejserjensen) [reply crates/modelardb_client/src/main.rs:224]: Fixed.

File: crates/modelardb_client/src/main.rs (hunk 2 above)
Comment (skejserjensen) [reply crates/modelardb_client/src/main.rs:215]: Other `Action`s are executed by `execute_action()` in `execute_command()` such as "FlushMemory" and "FlushEdge" so I am not entirely sure which function you mean that can only execute "CommandStatementUpdate".

File: crates/modelardb_client/src/main.rs (hunk 3 above)
Comment (skejserjensen) [reply crates/modelardb_client/src/main.rs:270]: Fixed.

File: crates/modelardb_client/src/main.rs (hunk 4 above)
Comment (skejserjensen) [reply crates/modelardb_client/src/main.rs:291]: Fixed.

File: crates/modelardb_client/src/main.rs (hunk 5 above)
Comment (skejserjensen) [reply crates/modelardb_client/src/main.rs:304]: Fixed.

**Review by CGodiksen — APPROVED** [review]

File: crates/modelardb_client/src/main.rs [hunk 14]
```diff
@@ -214,56 +196,67 @@ fn repl(
...
-) -> Result<(), Box<dyn Error>> {
-    if action_command_or_query.starts_with('\\') {
-        execute_command(runtime, flight_service_client, action_command_or_query)?;
-    } else if action_command_or_query.starts_with("SELECT")
+) {
+    let start_time = Instant::now();
+
+    let result = if action_command_or_query.starts_with('\\') {
+        execute_command(flight_service_client, action_command_or_query).await
+    } else if action_command_or_query.starts_with("INSERT")
```
Comment (chrthomsen) [inline crates/modelardb_client/src/main.rs:207]: does this mean that "INSERT" works, but "insert" does not?

File: crates/modelardb_client/src/main.rs [hunk 15]
```diff
@@ -272,38 +265,55 @@ fn execute_command(
...
             Ok(())
         }
+        // Flushes all data the server currently has in memory to disk.
+        "\\f" => execute_action(flight_service_client, "FlushMemory", "").await,
+        // Flushes all data the server currently has in memory and disk to the object store.
+        "\\F" => execute_action(flight_service_client, "FlushEdge", "").await,
+        // Print helpful information, explanations with \\ must be indented more to be aligned.
+        "\\h" => {
+            println!(
+                "CREATE DDL        Execute a CREATE TABLE or CREATE MODEL TABLE statement.\n\
```
Comment (chrthomsen) [inline crates/modelardb_client/src/main.rs:305]: is "CREATE DDL" enough in the help or should it say "CREATE [MODEL] TABLE tbl(col type, ...)" to provide more executable help?

File: crates/modelardb_client/src/main.rs (hunk 14 above)
Comment (skejserjensen) [reply crates/modelardb_client/src/main.rs:207]: Fixed through trim and uppercase.

File: crates/modelardb_client/src/main.rs (hunk 15 above)
Comment (skejserjensen) [reply crates/modelardb_client/src/main.rs]: Fixed.

**Review by chrthomsen — APPROVED** [review]

**Pull request merged**